core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -5200,11 +5186,50 @@ protected function searchMenu()
 		// Set the search value from the session
 		elseif ((string) $session['search'][$this->strTable]['value'] !== '')
 		{
-			$strPattern = "CAST(%s AS CHAR) REGEXP ?";
+			$searchValue = $session['search'][$this->strTable]['value'];
+
+			try
+			{
+				$this->Database->prepare("SELECT '' REGEXP ?")->execute($searchValue);
+			}
+			catch (DriverException $exception)
+			{
+				// Quote search string if it is not a valid regular expression
+				$searchValue = preg_quote($searchValue);
+			}
+
+			$strReplacePrefix = '';
+			$strReplaceSuffix = '';
+
+			// Decode HTML entities to make them searchable
+			if (empty($GLOBALS['TL_DCA'][$this->strTable]['fields'][$session['search'][$this->strTable]['field']]['eval']['decodeEntities']))
+			{
+				$arrReplace = array(
+					'&#35;' => '#',
+					'&#60;' => '<',
+					'&#62;' => '>',
+					'&lt;' => '<',
+					'&gt;' => '>',
+					'&#40;' => '(',
+					'&#41;' => ')',
+					'&#92;' => '\\\\',
+					'&#61;' => '=',
+					'&amp;' => '&',
+				);
+
+				$strReplacePrefix = str_repeat('REPLACE(', \count($arrReplace));
+
+				foreach ($arrReplace as $strSource => $strTarget)
+				{
+					$strReplaceSuffix .= ", '$strSource', '$strTarget')";
+				}
+			}
+
+			$strPattern = "$strReplacePrefix CAST(%s AS CHAR) $strReplaceSuffix REGEXP ?";
 
 			if (substr(Config::get('dbCollation'), -3) == '_ci')
 			{
-				$strPattern = "LOWER(CAST(%s AS CHAR)) REGEXP LOWER(?)";
+				$strPattern = "$strReplacePrefix LOWER(CAST(%s AS CHAR)) $strReplaceSuffix REGEXP LOWER(?)";
 			}
 
 			$fld = $session['search'][$this->strTable]['field'];
```

Comment:
Can we move this up to line 5190 and replace `$GLOBALS['TL_DCA'][$this->strTable]['fields'][$session['search'][$this->strTable]['field']]['eval']['decodeEntities']` with `$GLOBALS['TL_DCA'][$this->strTable]['fields'][$fld]['eval']['decodeEntities']` in line 5205?

Reply:
Done in 692f47f8bd04318446bfd1526ec676748ba28663